fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Memo.java

```diff
@@ -215,22 +229,19 @@ private void mergeGroup(Group srcGroup, Group dstGroup) {
                     reinsertExpression.getGroup().replaceBestExpression(reinsertExpression, existGroupExpression);
                     // existingGroupExpression merge the state of groupExpression
                     existGroupExpression.mergeGroupExpression(reinsertExpression);
+                } else {
+                    reinsertExpression.getGroup().deleteBestExpression(reinsertExpression);
+                    needMergeGroup.put(reinsertExpression.getGroup(), existGroupExpression.getGroup());
```

Comment:
```suggestion
                    existGroupExpression.getGroup().deleteBestExpression(reinsertExpression);
                    mergeGroupImpl(reinsertExpression.getGroup(), existGroupExpression.getGroup());
```

Reply:
1. Why delete existGroupExpression best expression？
2. We need do dstGroup.mergeGroup(srcGroup) first